src/BakedVertexAnimation/vertexAnimationBaker.ts

```diff
@@ -0,0 +1,158 @@
+import { AnimationRange } from "../Animations/animationRange";
+import { RawTexture } from "../Materials/Textures/rawTexture";
+import { Texture } from "../Materials/Textures/texture";
+import { Mesh } from "../Meshes/mesh";
+import { EncodeArrayBufferToBase64, DecodeBase64ToBinary } from "../Misc/stringTools";
+import { Scene } from "../scene";
+
+/**
+ * Class to bake vertex animation textures.
+ */
+export class VertexAnimationBaker {
+    private _scene: Scene;
+    private _mesh: Mesh;
+
+    /**
+     * Create a new VertexAnimationBaker object which can help baking animations into a texture.
+     * @param scene Defines the scene the VAT belongs to
+     * @param mesh Defines the mesh the VAT belongs to
+     * @param skeleton Defines the skeleton the VAT belongs to
+     */
+    constructor(scene: Scene, mesh: Mesh) {
+        this._scene = scene;
+        this._mesh = mesh;
+    }
+
+    /**
+     * Bakes the animation into the texture. This should be called once, when the
+     * scene starts, so the VAT is generated and associated to the mesh.
+     * @param ranges Defines the ranges in the animation that will be baked.
+     * @returns Float32Array

```

Comment:
Need better doc for the `@returns` value => there are a number of others below. Note that you don't have to document the protected/private methods, only the public ones.